src/services/prefix-manager/prefix-manager.service.ts

```diff
@@ -0,0 +1,25 @@
+import _ from "lodash"
+import { BotPrefixesEnum } from '../../enums/bot-prefixes.enum'
+
+export class PrefixManagerService {
+  private static _instance: PrefixManagerService
+
+  public static getInstance(): PrefixManagerService {
+    if(_.isNil(PrefixManagerService._instance)) {
+      PrefixManagerService._instance = new PrefixManagerService()
+    }
+    return PrefixManagerService._instance
+  }
+
+  private _getPrefix(message: string): string {
+    return message.split(/ +/)[0]
+  }
+
+  public setPrefix(message: string): string {
+    for (const prefix of Object.values(BotPrefixesEnum)) {
+      if (prefix === this._getPrefix(message)) return `${prefix} `
```

Comment:
Is it tested (not with jest, I mean on Discord)?

Reply:
Yes, it was tested! It works wonders! ^^